dsp-role/handler/src/test/scala/dsp/role/handler/RoleHandlerSpec.scala

```diff
@@ -0,0 +1,170 @@
+      for {
+        handler <- ZIO.service[RoleHandler]
+
+        newValue <- LangString.make("New Role Description", "en").toZIO
+
+        name1        <- RoleTestData.name1.toZIO
+        description1 <- RoleTestData.description1.toZIO
+        users1        = RoleTestData.users1
+        permission1  <- RoleTestData.permission1.toZIO
+
+        roleId <- handler.createRole(
+                    name1,
+                    description1,
+                    users1,
+                    permission1
+                  )
+
+        updatedRoleId <- handler.updateDescription(roleId, newValue)
+        retrievedRole <- handler.getRoleById(updatedRoleId)
+      } yield assertTrue(retrievedRole.name == name1) &&
+        assertTrue(retrievedRole.description == newValue) &&
+        assertTrue(retrievedRole.users == users1) &&
+        assertTrue(retrievedRole.permission == permission1)
+    },
+    // test("store a role and update its users") {
+    //   // implment adding users to the role
+    // },
+    test("store a role and update its name") {
+      for {
+        handler <- ZIO.service[RoleHandler]
+
+        newValue <- LangString.make("New Role Name", "en").toZIO
+
+        name1        <- RoleTestData.name1.toZIO
+        description1 <- RoleTestData.description1.toZIO
+        users1        = RoleTestData.users1
+        permission1  <- RoleTestData.permission1.toZIO
+
+        roleId <- handler.createRole(
+                    name1,
+                    description1,
+                    users1,
+                    permission1
+                  )
+
+        updatedRoleId <- handler.updateName(roleId, newValue)
+        retrievedRole <- handler.getRoleById(updatedRoleId)
+      } yield assertTrue(retrievedRole.name == newValue) &&
+        assertTrue(retrievedRole.description == description1) &&
+        assertTrue(retrievedRole.users == users1) &&
+        assertTrue(retrievedRole.permission == permission1)
+    },
+    test("store a role and update its permission") {
+      for {
+        handler <- ZIO.service[RoleHandler]
+
+        newValue <- Permission.make(Permission.Create).toZIO
+
+        name1        <- RoleTestData.name1.toZIO
+        description1 <- RoleTestData.description1.toZIO
+        users1        = RoleTestData.users1
+        permission1  <- RoleTestData.permission1.toZIO
+
+        roleId <- handler.createRole(
+                    name1,
+                    description1,
+                    users1,
+                    permission1
+                  )
+
+        updatedRoleId <- handler.updatePermission(roleId, newValue)
+        retrievedRole <- handler.getRoleById(updatedRoleId)
+      } yield assertTrue(retrievedRole.name == name1) &&
+        assertTrue(retrievedRole.description == description1) &&
+        assertTrue(retrievedRole.users == users1) &&
+        assertTrue(retrievedRole.permission == newValue)
+    }
+  ).provide(RoleRepoMock.layer, RoleHandler.layer)
+}
```

Comment:
I'm not sure but should there be tests to delete things? Delete a user from the role, delete the description of the role, or even delete a role?

Reply:
I'm not sure about the users, this was not yet clarified in the `Role` entity how will that work, but should the deletion of the description be possible if its required? Added missing test for delete method.